pkg/resourcemonitor/noderesourcesaggregator.go

```diff
@@ -41,30 +40,18 @@ type nodeResources struct {
 	perNUMACapacity map[int]map[v1.ResourceName]int64
 	// mapping: resourceName -> resourceID -> nodeID
 	resourceID2NUMAID map[string]map[string]int
-	cpuInfo           *cpus.CPUs
-	numaInfo          *numa.Nodes
-	numaDists         *distances.Distances
+	topo              *ghw.TopologyInfo
 }
 
 type resourceData struct {
 	allocatable int64
 	capacity    int64
 }
 
-func NewResourcesAggregator(sysfsPath string, podResourceClient podresourcesapi.PodResourcesListerClient) (ResourcesAggregator, error) {
+func NewResourcesAggregator(sysfsRoot string, podResourceClient podresourcesapi.PodResourcesListerClient) (ResourcesAggregator, error) {
 	var err error
 
-	cpuInfo, err := cpus.NewCPUs(sysfsPath)
-	if err != nil {
-		return nil, err
-	}
-
-	nodes, err := numa.NewNodesFromSysFS(sysfsPath)
-	if err != nil {
-		return nil, err
-	}
-
-	dists, err := distances.NewDistancesFromSysfs(sysfsPath)
+	topo, err := ghw.Topology(ghw.WithChroot(sysfsRoot))
```

Comment:
Reading through [this](https://github.com/jaypipes/ghw#overriding-the-root-mountpoint-ghw-uses), I've noticed that WithChroot is used to override root mountpoint and not sysfs like we are doing in this case. Because of this we will end up using` /host-sys/sys` as our `sysfs` and cpu resources will not computed/accounted correctly.

Reply:
Right! Thanks for spotting this. This is a limitation (pretty fair however) of how ghw works. I think we can sidestep this problem changing the way we mount the host sysfs and procfs inside the NFD container. Would that work for you?
If so I'll add the patch to change the mountpoints in this PR.